pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -315,12 +297,8 @@ func (b *Build) BuildPod(images BuildPodImages, buildContext BuildContext) (*cor
 			layersMount,
 			workspaceVolume,
 			homeMount,
-		}, func() []corev1.VolumeMount {
-			if platformAPILessThan07 {
-				return cacheVolumes
-			}
-			return []corev1.VolumeMount{}
-		}()),
+		},
+			[]corev1.VolumeMount{}),
```

Comment:
The `volumeMounts` func does the same thing as `args`, so passing in an empty slice is noop

```suggestion
		}),
```